javaslang-benchmark/src/test/java/javaslang/benchmark/collection/ArrayBenchmark.java

```diff
@@ -17,21 +21,146 @@ public static void main(String... args) { /* main is more reliable than a test *
 
         public Integer[] ELEMENTS;
 
+        java.util.ArrayList<Integer> javaMutable;
+        fj.data.Array<Integer> fjavaPersistent;
+        javaslang.collection.Array<Integer> slangPersistent;
+
         @Setup
         public void setup() {
             ELEMENTS = getRandomValues(CONTAINER_SIZE, 0);
+
+            assertEquals(javaMutable, null);
+            javaMutable = new ArrayList<>(CONTAINER_SIZE);
+            Collections.addAll(javaMutable, ELEMENTS);
+            assertEquals(javaMutable.size(), CONTAINER_SIZE);
+
+            assertEquals(fjavaPersistent, null);
+            fjavaPersistent = Array.array(ELEMENTS);
+            assertEquals(fjavaPersistent.length(), CONTAINER_SIZE);
+
+            assertEquals(slangPersistent, null);
+            slangPersistent = javaslang.collection.Array.of(ELEMENTS);
+            assertEquals(slangPersistent.size(), CONTAINER_SIZE);
+        }
+    }
+
+    public static class Head extends Base {
+        @Benchmark
+        public Object java_mutable() { return javaMutable.get(0); }
+
+        @Benchmark
+        public Object fjava_persistent() { return fjavaPersistent.get(0); }
+
+        @Benchmark
+        public Object slang_persistent() { return slangPersistent.head(); }
+    }
+
+    public static class Tail extends Base {
+        @State(Scope.Thread)
+        public static class Initialized {
+            java.util.ArrayList<Integer> javaMutable;
+            javaslang.collection.Array<Integer> slangPersistent = javaslang.collection.Array.empty();
+
+            @Setup(Level.Invocation)
+            public void initializeMutable(Base state) {
+                assertEquals(javaMutable, null);
+                javaMutable = new java.util.ArrayList<>(state.CONTAINER_SIZE);
+                Collections.addAll(javaMutable, state.ELEMENTS);
+                assertEquals(javaMutable.size(), state.CONTAINER_SIZE);
+
+                if (slangPersistent.isEmpty()) {
+                    slangPersistent = slangPersistent.appendAll(Iterator.of(state.ELEMENTS));
+                    assertEquals(slangPersistent.size(), state.CONTAINER_SIZE);
+                }
+            }
+
+            @TearDown(Level.Invocation)
+            public void tearDown() {
+                javaMutable = null;
+            }
+        }
+
+        @Benchmark
+        public Object java_mutable(Initialized state) {
+            state.javaMutable.remove(0);
```

Comment:
since this modifies the state, we have to recreate the original `ArrayList` every time